File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/splash/MapsWelcomeFragment.java
package ua.com.foxminded.locationtrackera.ui.splash;

import android.view.View;

import ua.com.foxminded.locationtrackera.R;
import ua.com.foxminded.locationtrackera.util.SafeNavigation;

public class MapsWelcomeFragment extends WelcomeFragment {

    @Override
    public void onClick(View view) {
        if (view == binding.welcomeRegisterBtn) {
            SafeNavigation.navigate(binding.getRoot(), R.id.mapsWelcomeFragment,
                    R.id.nav_from_mapsWelcomeFragment_to_mapsRegistrationFragment);
        } else if (view == binding.welcomeLoginTxt) {
            SafeNavigation.navigate(binding.getRoot(), R.id.mapsWelcomeFragment,
                    R.id.nav_from_mapsWelcomeFragment_to_mapsLoginFragment);
        }
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/splash/MapsWelcomeFragment.kt
package ua.com.foxminded.locationtrackera.ui.splash

import android.view.View
import ua.com.foxminded.locationtrackera.R
import ua.com.foxminded.locationtrackera.util.SafeNavigation

class MapsWelcomeFragment : WelcomeFragment() {

    override fun onClick(view: View) {
        when {
            view === binding.welcomeRegisterBtn -> SafeNavigation.navigate(
                binding.root, R.id.mapsWelcomeFragment,
                R.id.nav_from_mapsWelcomeFragment_to_mapsRegistrationFragment
            )
            view === binding.welcomeLoginTxt -> SafeNavigation.navigate(
                binding.root, R.id.mapsWelcomeFragment,
                R.id.nav_from_mapsWelcomeFragment_to_mapsLoginFragment
            )
        }
    }
}